### DISCUSSION TITLE
bigint overflow

### DISCUSSION THREAD

**Discussion by tanyaofei** [fastapi/fastapi#7543] [Questions] [answered]
### Describe the bug

Routing can not be return bigint value correctly

### To Reproduce

1. Create a file with:

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/{test_id}")
def read_root(test_id: int):
    return test_id
```

3. Open the browser and call the endpoint `/1249958025205260288`.
4. It returns  `1249958025205260300 `.
5. But I expected it to return `1249958025205260288`.

### Environment

- OS: macOS & Linux
- FastAPI Version : 0.52.0
- Python Version: 3.7.2 & 3.7.4

**Comment by Dustyposa** [comment, marked as answer]
I think this problem caused by js, not python,even fasapi.
In js:
You `console.log(1249958025205260288)`, you can see it consoled :
```
1249958025205260300
```
Not `1249958025205260288`.

**Comment by tanyaofei** [comment]
yes, this problem caused by js

**Comment by phy25** [comment]
You can close the issue since it's really not a fastapi issue.

**Comment by tanyaofei** [comment]
sure, have a nice day

在 2020年4月22日，13:46，Henry Pan <notifications@github.com> 写道：

﻿

You can close the issue since it's really not a fastapi issue.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue @tanyaofei  :+1:
